### ISSUE TITLE
Bundler/GemComment vs Bundler/OrderedGems

### ISSUE THREAD

**Issue by coding-bunny** [rubocop-hq/rubocop#9354]
## Expected behavior

I expect these gems to work in tandem with one another.
Ideally the `Bundler/GemComment` uses the `OnlyFor: ['version_specifiers']` configuration

## Actual behavior

At the moment enabling both Cops kind of makes the `Bundler/OrderedGems` obsolete.
Because the `Bundler/GemComment` requires a comment above each gem entry, the `Bundler/OrderedGems` is no longer able to determine whether gems are in alphabetical order.

## Steps to reproduce the problem

Create a following Gemfile:
```Gemfile
# frozen_string_literal: true

source 'https://rubygems.org'
git_source(:github) { |repo| "git@github.com:#{repo}.git" }

# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
gem 'rails', '6.1.1'

# Environment variable support, needs to be loaded before anything
gem 'dotenv-rails', require: 'dotenv/rails-now'

# Used to communicate with Braintree, our payment gateway provider
gem 'braintree'
# Used to speed up development boot time
gem 'bootsnap', '>= 1.4.2', require: false
```

I expect the `Bundler/OrderedGems` to report the following:

* Bootsnap should be before braintree

But it doesn't because of the comments injected. Removing the comments makes it work IF there's no linebreaks between gems.
The same applies if they are inside a block.

## RuboCop version

```
1.8.0 (using Parser 3.0.0.0, rubocop-ast 1.4.0, running on ruby 2.7.2 x86_64-linux)
  - rubocop-performance 1.9.2
  - rubocop-rake 0.5.1
```

**Comment by anoymouscoder** [comment]
@marcandre   
I can help solve this problem.

**Comment by anoymouscoder** [comment]
@marcandre  According to https://docs.rubocop.org/rubocop/cops_bundler.html#examples-4
```ruby
# good only if TreatCommentsAsGroupSeparators is true
# For code quality
gem 'rubocop'
# For tests
gem 'rspec'
```
This style currently considered good, so before I solve it, I think we need to agree on the type of `good`.

**Comment by marcandre** [comment]
I think the issue is that even with `TreatCommentsAsGroupSeparators: false`, it does not generate an issue if there is an empty line and a comment separating gems in the wrong order.

**Comment by anoymouscoder** [comment]
@marcandre   This issue clearly describes not what you mean. As for the scene what you said, [[Fix #4004] Allow not treating comment lines as gem group separators in `Bundler/OrderedGems` cop](https://github.com/rubocop-hq/rubocop/commit/fecc0d3a1e61c8f046b83d9fc653aa9b1bf194ed) already solved.

**Comment by anoymouscoder** [comment]
@marcandre
I submitted a [pr](https://github.com/rubocop-hq/rubocop/pull/9404), PTAL

**Comment by coding-bunny** [comment]
Closing this because it's working how it should, and I'm a bunny-brain.
